vars/artifactSetVersion.groovy

```diff
@@ -70,7 +89,13 @@ import groovy.text.SimpleTemplateEngine
     /** Defines the template for the timestamp which will be part of the created version. */
     'timestampTemplate',
     /** Defines the template for the automatic version which will be created. */
-    'versioningTemplate'
+    'versioningTemplate',
+    /** Controls which protocol is used for performing push operation to remote repo.
+      * Required credentials needs to be configured ('gitSshKeyCredentialsId'/'TBD').
+      * Push is only performed in case 'commitVersion' is set to 'true'.
+      * @possibleValues 'SSH', 'HTTPS', 'NONE'
+      */
+    'gitPushMode',
```

Comment:
why don't we just check for the credentials instead of adding another parameter?

Reply:
This way in more explicit. Basically I think we should "by default" provide defaults. Would be possible for credentialIds, but if we overload (-> having a credentialsId does not only mean having the credentialsId, but does also mean we deploy with this and that protocol) we cannot provide a default value here. In essence: I think being as explict as possible here is an advantage. Also: during setting up scenarios or troubleshooting it is more easy and straight forward to simply toggle a mode rather than having to remove / add again some credential ids. IMO having more parameters is fine here since having the parameters explicit without "transitive" meaning simplifies our life.

Reply:
I would not do so but anyway.

If adding a parameter we should not limit this to `pushMode`. It might also apply for clones etc. going forward in case this is really important to users.

So far we relied on ssh in all cases ...